eng/common/scripts/stress-testing/find-all-stress-packages.ps1

```diff
@@ -62,6 +69,31 @@ function GetUsername() {
     return $stressUser.ToLower()
 }
 
+function ScenariosMatrixGeneration(
+    [string]$matrixFilePath
+) {
+    $yamlConfig = Get-Content $matrixFilePath -Raw
+    $matrix = GenerateMatrix (GetMatrixConfigFromYaml $yamlConfig) "sparse"
+    $serializedMatrix = SerializePipelineMatrix $matrix
+    $prettyMatrix = $serializedMatrix.pretty | ConvertFrom-Json -AsHashtable
+
+    $scenariosMatrix = @()
+    foreach($permutation in $prettyMatrix.GetEnumerator()) {
+        $entry = @{}
+        $entry.Name = $permutation.key
+        foreach ($param in $permutation.value.GetEnumerator()) {
+            $entry.add($param.key, $param.value)
+        }
+        $scenariosMatrix += $entry
+    }
+
+    $valuesYaml = Get-Content (Join-Path $matrixFilePath '../values.yaml') -Raw
+    $values = $valuesYaml | ConvertFrom-Yaml
+
+    $values.Scenarios += $scenariosMatrix
```

Comment:
Do you need to create `Scenarios` if it doesn't exist?

Reply:
Also I think we should throw if `Scenarios` already exists, because it means someone could be using the legacy list of strings format which is incompatible.